lib/validation.js

```diff
@@ -76,32 +76,40 @@ function validateParam (validatorFunction, request, paramName) {
 
 function validate (context, request) {
   var params = validateParam(context[paramsSchema], request, 'params')
+  var schemaErrorFormatter = (context.schemaErrorFormatter || schemaErrorsText).bind(request)
+
   if (params) {
-    return wrapValidationError(params, 'params')
+    return wrapValidationError(params, 'params', schemaErrorFormatter)
   }
   var body = validateParam(context[bodySchema], request, 'body')
   if (body) {
-    return wrapValidationError(body, 'body')
+    return wrapValidationError(body, 'body', schemaErrorFormatter)
   }
   var query = validateParam(context[querystringSchema], request, 'query')
   if (query) {
-    return wrapValidationError(query, 'querystring')
+    return wrapValidationError(query, 'querystring', schemaErrorFormatter)
   }
   var headers = validateParam(context[headersSchema], request, 'headers')
   if (headers) {
-    return wrapValidationError(headers, 'headers')
+    return wrapValidationError(headers, 'headers', schemaErrorFormatter)
   }
   return null
 }
 
-function wrapValidationError (result, dataVar) {
+function wrapValidationError (result, dataVar, schemaErrorFormatter) {
   if (result instanceof Error) {
-    return result
+    return Object.assign(result, { statusCode: 400 })
+  }
+
+  try {
+    return Object.assign(schemaErrorFormatter(result, dataVar), {
```

Comment:
`Object.assign` should be less performant than assign the `validation` and `validationContext` individually